src/collections.rs

```diff
@@ -201,6 +263,109 @@ impl TestCase {
+impl TestCaseBuilder {
+    /// Creates a new TestCaseBuilder for a successful `TestCase`
+    pub fn success(name: &str, time: Duration) -> Self {
+        TestCaseBuilder {
+            name: name.into(),
+            time,
+            result: TestResult::Success,
+            classname: None,
+            system_out: None,
+            system_err: None,
+        }
+    }
+
+    /// Set the `classname` for the `TestCase`
+    pub fn set_classname(&mut self, classname: &str) -> &mut Self {
+        self.classname = Some(classname.to_owned());
+        self
+    }
+
+    /// Set the `system_out` for the `TestCase`
+    pub fn set_system_out(&mut self, system_out: &str) -> &mut Self {
+        self.system_out = Some(system_out.to_owned());
+        self
+    }
+
+    /// Set the `system_err` for the `TestCase`
+    pub fn set_system_err(&mut self, system_err: &str) -> &mut Self {
+        self.system_err = Some(system_err.to_owned());
+        self
+    }
+
+    /// Creates a new TestCaseBuilder for an erroneous `TestCase`
+    ///
+    /// An erroneous `TestCase` is one that encountered an unexpected error condition.
+    pub fn error(name: &str, time: Duration, type_: &str, message: &str) -> Self {
+        TestCaseBuilder {
+            name: name.into(),
+            time,
+            result: TestResult::Error {
+                type_: type_.into(),
+                message: message.into(),
+            },
+            classname: None,
+            system_out: None,
+            system_err: None,
+        }
+    }
+
+    /// Creates a new TestCaseBuilder for a failed `TestCase`
+    ///
+    /// A failed `TestCase` is one where an explicit assertion failed
+    pub fn failure(name: &str, time: Duration, type_: &str, message: &str) -> Self {
+        TestCaseBuilder {
+            name: name.into(),
+            time,
+            result: TestResult::Failure {
+                type_: type_.into(),
+                message: message.into(),
+            },
+            classname: None,
+            system_out: None,
+            system_err: None,
+        }
+    }
+
+    /// Creates a new TestCaseBuilder for an ignored `TestCase`
+    ///
+    /// An ignored `TestCase` is one where an ignored or skipped
+    pub fn skipped(name: &str) -> Self {
+        TestCaseBuilder {
+            name: name.into(),
+            time: Duration::zero(),
+            result: TestResult::Skipped,
+            classname: None,
+            system_out: None,
+            system_err: None,
+        }
+    }
+
+    /// Build and return a [`TestCase`](struct.TestCase.html) object based on the data stored in this TestCaseBuilder object.
+    pub fn build(&self) -> TestCase {
+        TestCase {
+            name: self.name.clone(),
+            time: self.time.clone(),
+            result: self.result.clone(),
+            classname: self.classname.clone(),
+            system_out: self.system_out.clone(),
+            system_err: self.system_err.clone(),
+        }
```

Comment:
Build could then just return the inner instance and consuming it in the process.

Reply:
Changed this to
```rust
pub fn build(&self) -> TestSuite {
    self.testsuite.clone()
}
```

Which doesn't consume the inner instance but returns a clone instead.
To make the inner instance consumable here it would be necessary to pass `self` instead of `&self`, but that would impact usability of the builder. Have a look at [Rust builder docs](https://doc.rust-lang.org/1.0.0/style/ownership/builders.html) for more details.